src/rgw/rgw_crypt.cc

```diff
@@ -726,7 +751,7 @@ int RGWGetObj_BlockDecrypt::fixup_range(off_t& bl_ofs, off_t& bl_end) {
 int RGWGetObj_BlockDecrypt::process(bufferlist& in, size_t part_ofs, size_t size)
 {
   bufferlist data;
-  if (!crypt->decrypt(in, 0, size, data, part_ofs)) {
+  if (!crypt->decrypt(in, 0, size, data, part_ofs, null_yield)) {
```

Comment:
we'll want to pass a real yield context into `decrypt()` here

in `RGWGetObj_ObjStore_S3::get_decrypt_filter()`, you can pass `s->yield` into `RGWGetObj_BlockDecrypt`'s constructor

Reply:
> we'll want to pass a real yield context into `decrypt()` here
> 
> in `RGWGetObj_ObjStore_S3::get_decrypt_filter()`, you can pass `s->yield` into `RGWGetObj_BlockDecrypt`'s constructor

Done. I pass `optional_yield`(`s->yield`) into RGWGetObj_BlockDecrypt's constructor. And this `null_yield` was replaced with `y`.